pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -496,6 +505,7 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		}
 
 		cdLog.Debug("cluster is already installed, no processing of provision needed")
+		r.cleanupInstallPod(cd, cdLog)
```

Comment:
The clusterdeployment controller should not be doing anything with the install pods. It should be the clusterprovision controller that deletes the install pod when the provision is successful.